core_types/src/cas/content.rs

```diff
@@ -33,7 +34,7 @@ pub trait AddressableContent {
 
     /// restore/deserialize the original struct/type from serialized Content
     /// the default implementation covers anything that implements From<JsonString> for Foo
-    fn from_content(content: &Content) -> Self
+    fn try_from_content(content: &Content) -> Result<Self, HolochainError>
```

Comment:
You say that we can't use `TryFrom<T>` because `try_from()` returns `Self` and thus would prevent AddressableContent from being made into a trait object. This function has `Self` wrapped in a Result. Doesn't that prevent trait objects as well? Or does `Result` store its content on the heap? @thedavidmeister

Reply:
@lucksus hmmmm 🤔 i think i need @AshantiMutinta to weigh in on this one!

this definitely works now while `TryFrom` definitely broke when i tried it...

Reply:
The self in the result definitely stops it from being a trait object sadly. I think it is fine because there is no AddressableContent trait object being used for now and if we ever need one we could refactor it in.(The issue would be if we would want that AddressableContent in a trait that needs AC to be a trait object). This seems fine to me!

Reply:
So what made the TryFrom solution break? If it was not this, maybe we could use TryFrom? But I don't want to halt merging this. Feel free to either investigate what's going on here, or merge, @thedavidmeister